### PR TITLE
Added SIMBODY_EXTRA_CMAKE_ARGS to dependencies/ (#3455)

### PR DESCRIPTION
**Pull request by adamkewley**
Fixes issue #3455 

### Brief summary of changes

- Adds `SIMBODY_EXTRA_CMAKE_ARGS` to `dependencies/CMakeLists.txt`
- This enables customizing the simbody build when creating a superbuild of the dependencies

### Testing I've completed

- "Works on my machine"
- "Probably simple enough to not cause problems"

### CHANGELOG.md (choose one)

- updated.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3456)

### CODE REVIEW

File: dependencies/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,12 @@ include(ExternalProject)
 include(CMakeParseArguments)
 include(CMakeDependentOption)
 
+set(
+    SIMBODY_EXTRA_CMAKE_ARGS ""
+    CACHE STRING
+    "extra args to forward when configuring Simbody with CMake. The format must be: VARNAME1:TYPE1=VALUE1;VARNAME2:TYPE2=VALUE2"
```
Comment (tkuchida) [inline dependencies/CMakeLists.txt:16]: (optional) Point to a list of possible VARNAMEs or provide one example (e.g., the arguments you are using) to hopefully make this more accessible to users?

**Review by tkuchida — APPROVED** [review]
One optional suggestion below 🐄

File: dependencies/CMakeLists.txt (hunk 1 above)
Comment (adamkewley) [reply dependencies/CMakeLists.txt]: I've clarified the documentation slightly in 1fce59e

**Comment by adamkewley** [comment]
Due to another PR being merged, this is now a non-mergeable PR - I'll perform a merge commit to clean it up

**Pull request merged**